### ISSUE TITLE
Bug: `infer_bbox_shape3d` / `bbox_to_mask3d` accept `(B, N, 8, 3)` through `validate_bbox3d` and then crash or return wrong-shaped values

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4248]
`validate_bbox3d` documents and accepts rank-4 `(B, N, 8, 3)` input (it reshapes to `(B*N, 8, 3)` internally), but its two callers in the same module do not reshape: `infer_bbox_shape3d` runs `torch.index_select(boxes, 1, ...)` and `bbox_to_mask3d` indexes `boxes[:, 4, 2]` on the original tensor, so the box axis is read as the vertex axis. This is the 3D twin of #4180, which #4218 fixed for `infer_bbox_shape` / `bbox_to_mask` by rejecting rank 4 with `ShapeError`.

Reproduced 2026-09-05 at `1eea5835` (main after #4245), PyTorch 2.14.0, Python 3.11, CPU and MPS.

## Reproduction

Run from the checkout root:

```python
import torch
from kornia.geometry.bbox import validate_bbox3d, infer_bbox_shape3d, bbox_to_mask3d

cuboid = torch.tensor([[[1., 1., 1.], [3., 1., 1.], [3., 3., 1.], [1., 3., 1.],
                        [1., 1., 2.], [3., 1., 2.], [3., 3., 2.], [1., 3., 2.]]])
for n in (1, 3, 8):
    boxes = cuboid.expand(n, 8, 3)[None].contiguous()  # (1, n, 8, 3)
    print(n, "validate:", validate_bbox3d(boxes))
    for name, call in (("infer", lambda: infer_bbox_shape3d(boxes)), ("mask", lambda: bbox_to_mask3d(boxes, (4, 5, 5)))):
        try:
            out = call()
            print("  ", name, [tuple(t.shape) for t in out] if isinstance(out, tuple) else tuple(out.shape))
        except Exception as error:
            print("  ", name, type(error).__name__, str(error)[:70])
```

## Observed

```text
1 validate: True
   infer RuntimeError INDICES element is out of DATA bounds, id=1 axis_dim=1
   mask IndexError index 4 is out of bounds for dimension 1 with size 1
3 validate: True
   infer RuntimeError INDICES element is out of DATA bounds, id=5 axis_dim=3
   mask IndexError index 4 is out of bounds for dimension 1 with size 3
8 validate: True
   infer [(1, 3), (1, 3), (1, 3)]
   mask RuntimeError The size of tensor a (4) must match the size of tensor b (3) at non-singleton dimension 3
```

With `N = 8` the vertex-axis reads succeed by coincidence and `infer_bbox_shape3d` returns three `(1, 3)` tensors, one value per coordinate instead of one per box; eight distinct boxes give `depths = [5, 5, 5]` where the flattened `(8, 8, 3)` input gives eight 2s. On MPS the `N = 1` case raises `torch.AcceleratorError` instead of `RuntimeError`; the outcome is the same. `N = 5, 7, 9` were also swept: all raise.

## Expected

Parity with the 2D helpers after #4218: reject rank 4 with `ShapeError` naming the documented `(B, 8, 3)` contract, or flatten it the way `validate_bbox3d` does. Either is a focused fix; the batch-4c convention blocks document the current behavior and carry a wart pin (`test_wart_rank4_input_passes_validate_bbox3d_then_breaks_the_3d_helpers_4248`, `N` in `{1, 8}`) plus a strict xfail asserting the `ShapeError` outcome (`test_convention_rank4_input_is_rejected_by_the_3d_helpers_4248`), so a fix that flattens instead must update that xfail. `kornia/geometry/transform/crop3d.py` calls both helpers with rank-3 input and is unaffected.

## Classification

Broken output: a shape the module's own validator admits crashes with an unrelated indexing error, and for one batch size silently returns wrong-shaped values. Idiom 3 (the accepted shape contract is wider than the implementation's indexing assumptions).

## Disposition

n/a — single symbol

Found during the bbox/boxes conventions audit batch 4c (#4020); the audit changes no runtime behavior.

Posted on behalf of @ducha-aiki by Claude (Fable 5.1).
